BookingsApi/BookingsApi.Domain/Hearing.cs

```diff
@@ -675,12 +682,32 @@ public void UpdateStatus(BookingStatus newStatus, string updatedBy, string cance
 
         public ParticipantBase GetJudge()
         {
-            var judge = Participants?.FirstOrDefault(p => p.HearingRole.UserRole.IsJudge);
+            var judge = Participants?.FirstOrDefault(p => p.HearingRole?.UserRole?.IsJudge ?? false);
             var judiciaryJudge = JudiciaryParticipants?.FirstOrDefault(p => p.HearingRoleCode == JudiciaryParticipantHearingRoleCode.Judge);
 
             return (ParticipantBase)judge ?? judiciaryJudge;
         }
         
+
+        public void UpdateBookingStatusJudgeRequirement()
```

Comment:
Looking at how many changes were made to invoke this method, it feels like each domain method that causes a change (i.e. remove participant) could call this method, keeping the responsibility of updating the status from changes like this internal. Makes for a cleaner change too now that I've taken a fresh look.